### DISCUSSION TITLE
How to use a VPN just for the sandbox?

### DISCUSSION THREAD

**Discussion by froggy2090** [netblue30/firejail#6502] [Q&A] [answered]
Hi everyone,

I need some help with Firejail. I'm trying to figure out if there's a way to connect a VPN within Firejail. Specifically, I want to run an alternate network interface using `--net`, and then use Netfilter to route all traffic through Tor from that alternate network interface. 

The goal is to connect a VPN through this alternate interface. However, I don't have advanced networking knowledge, so I'm struggling a bit. I’m also considering the option of connecting the VPN directly from the host system if that simplifies the process but i am not sure if it is the most convenient way to do it.

Any guidance or advice on how to achieve this setup would be greatly appreciated!

Thank you in advance!

**Comment by amano-kenji** [comment, marked as answer]
Firejail won't set up complex network infrastructure. You will need to write a few init services that set up network namespaces with VPN capabilities.

Run `firejail --netns=vpn1` or `firejail --netns=vpn2`.

I applied https://www.wireguard.com/netns/ to my own system. You can also redirect a traffic from a network namespace to tor through firewall rules.
